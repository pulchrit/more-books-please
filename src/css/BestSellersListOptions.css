.best-sellers-list-options {
    width: 90%;
    max-width: 500px;
    margin: 3% auto;
    display: flex;
    flex-direction: column;
}

.subhead {
    text-align: center; 
    font-size: 1.3em;
}

.list-button {
    background-color: #30490A;
    border-color: #30490A;
    border-radius: 3px;
    padding: 30px;
    margin: 10px;
    cursor: pointer;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.list-button a:link,
.list-button a:visited,
.list-button a:hover,
.list-button a:focus,
.list-button a:active {
    color: white;
    font-family: 'Crimson Text', serif;
    line-height: 1;
    font-weight: 700;
    font-size: 1.6em;
    text-decoration: none;
    margin: 0;
}

.directions {
    padding: 0 30px;
    margin-top: 0;
    font-size: 1.1em;
    letter-spacing: .65px;
    line-height: 1.6;
}