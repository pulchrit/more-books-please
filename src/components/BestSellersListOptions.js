import React from 'react';
import {Link} from 'react-router-dom';
import '../css/BestSellersListOptions.css';

export default class BestSellersListOptions extends React.Component {

    render() {

        // Save error from state to error variable.
        const {error} = this.props;
        
        const nameButtons = this.props.listNames.map((name, index) =>  
            <button className="list-button" key={index}>
                <Link to={`/topBooks/${name.list_name_encoded}`}>
                        {name.display_name}
                </Link>
            </button>
        );
            
        return (
            <section className="best-sellers-list-options">
                
                {/* If there is an error, render it, otherwise 'display' empty string. */}
                {error ? <p className='error' role='alert'>{error.message}</p> : ''}

                <h2 className="subhead">Find more food for your hungry little bookworms.</h2>
                <p className="directions">Choose from The New York Times Bestsellers Lists below
                    to see the most recent best sellers in each children's book list.</p>
                    {nameButtons}
            </section>
        );
    } 
}
