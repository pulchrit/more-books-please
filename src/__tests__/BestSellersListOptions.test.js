import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import BestSellersListOptions from '../components/BestSellersListOptions';

describe ("BestSellersListOptions", () => {

    const listNames= [
        {
            display_name: "Children's Picture Books",
            list_name_encoded: 'picture-books'
        },
        {
            display_name: "Children's Chapter Books",
            list_name_encoded: 'chapter-books'
        },
        {
            display_name: "Children's Middle Grade Books",
            list_name_encoded: 'childrens-middle-grade'
        }
    ];
            
    const error = null;

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <BestSellersListOptions 
            listNames={listNames}
            error={error}/>
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <BestSellersListOptions 
                listNames={listNames}
                error={error}
            />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

})


