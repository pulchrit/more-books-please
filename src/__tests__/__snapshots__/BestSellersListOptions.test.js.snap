// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestSellersListOptions renders UI as expected 1`] = `
<section
  className="best-sellers-list-options"
>
  
  <h2
    className="subhead"
  >
    Find more food for your hungry little bookworms.
  </h2>
  <p
    className="directions"
  >
    Choose from The New York Times Bestsellers Lists below to see the most recent best sellers in each children's book list.
  </p>
  <button
    className="list-button"
  >
    <a
      href="/topBooks/picture-books"
      onClick={[Function]}
    >
      Children's Picture Books
    </a>
  </button>
  <button
    className="list-button"
  >
    <a
      href="/topBooks/chapter-books"
      onClick={[Function]}
    >
      Children's Chapter Books
    </a>
  </button>
  <button
    className="list-button"
  >
    <a
      href="/topBooks/childrens-middle-grade"
      onClick={[Function]}
    >
      Children's Middle Grade Books
    </a>
  </button>
</section>
`;
